### PR TITLE
Feature/group chat

### PR DESCRIPTION
**Pull request by mr-rooftop**
Switched from user_data to chat_data.
Changed multiple handlings to make group chats work.
Implemented MessageQueue to mitigate Telegram Group Chat Rate Limiting

### CODE REVIEW

File: bot/jobs.py [hunk 1]
```diff
@@ -289,21 +292,26 @@ def check_thorchain_catch_up_status(context, node_address):
...
+        return
     except Exception as e:
         logger.exception(e)
         return
 
     if node_data['is_catching_up'] != is_currently_catching_up:
         try:
             block_height = get_latest_block_height(node_data['ip_address'])
+        except (Timeout, ConnectionError):
+            logger.info(f"Timeout or Connection error with {node_data['ip_address']}")
```
Comment (mbartn) [inline bot/jobs.py:313]: i think `info` it's not appropriate level here as this is not normal application behavior, i would call it `warn` or `debug`. The same in other places

File: bot/service/thorchain_network_service.py [hunk 2]
```diff
@@ -32,8 +31,14 @@ def is_thorchain_catching_up(node_ip=None) -> bool:
 def is_midgard_api_healthy(node_ip) -> bool:
     try:
         get_request_json_thorchain(url_path=":8080/v1/health", node_ip=node_ip)
+    except (Timeout, ConnectionError):
+        logger.info(f"Timeout or Connection error with {node_ip}")
+        return False
     except Exception as e:
-        logger.exception(e)
+        if e.args[0].status_code == 502:
```
Comment (mbartn) [inline bot/service/thorchain_network_service.py:38]: Imo this is dangerous and can kill the bot, what if there is no args or status_code? Is it possible?

File: bot/thornode_bot.py [hunk 3]
```diff
@@ -27,81 +29,84 @@ def setup_existing_users(dispatcher):
...
+                'chat_data': dispatcher.chat_data[chat_id]
             })
 
     for chat_id in blocked_ids:
-        logger.warning(f"Telegram user {str(chat_id)} blocked me or is "
-                       f"not Admin anymore; removing him from the user list")
+        logger.warning(f"Telegram chat {str(chat_id)} blocked me or is "
+                       f"not Admin anymore; removing it from the chat list")
+
+        if chat_id > 0:
```
Comment (mbartn) [inline bot/thornode_bot.py:65]: same as earlier

**Review by mbartn — APPROVED** [review]
tiny changes suggested, nice solution with queue 👍

File: bot/thornode_bot.py [hunk 4]
```diff
@@ -27,81 +29,84 @@ def setup_existing_users(dispatcher):
 
     blocked_ids = []
 
-    for chat_id in dispatcher.user_data.keys():
-        if ALLOWED_USER_IDS != 'ALL' and chat_id not in ALLOWED_USER_IDS:
+    for chat_id in dispatcher.chat_data.keys():
+        if ALLOWED_USER_IDS != 'ALL' and chat_id > 0 and chat_id not in ALLOWED_USER_IDS:
```
Comment (mbartn) [inline bot/thornode_bot.py:33]: I would assign `chat_id > 0` to a variable so devs know what does it mean, something like `is_private_chat` (?)

File: bot/service/thorchain_network_service.py (hunk 2 above)
Comment (mr-rooftop) [reply bot/service/thorchain_network_service.py:38]: Pretty sure those fields must exist, but added more safety checks thx :+1:

File: bot/jobs.py (hunk 1 above)
Comment (mr-rooftop) [reply bot/jobs.py:313]: Wanted to get rid of the long error stack traces for errors that we expect to happen, that's why I removed the logger.error() stuff. 
But true, that's not the correct usage.

**Comment by mr-rooftop** [comment]
Thank you for the constructive review

**Pull request merged**